stan/math/prim/mat/prob/multi_gp_log.hpp

```diff
@@ -43,61 +33,20 @@ namespace stan {
                  const Eigen::Matrix
                  <T_covar, Eigen::Dynamic, Eigen::Dynamic>& Sigma,
                  const Eigen::Matrix<T_w, Eigen::Dynamic, 1>& w) {
-      static const char* function("multi_gp_log");
-      typedef typename boost::math::tools::promote_args<T_y, T_covar, T_w>::type
-        T_lp;
-      T_lp lp(0.0);
-
-
-      check_positive(function, "Kernel rows", Sigma.rows());
-      check_finite(function, "Kernel", Sigma);
-      check_symmetric(function, "Kernel", Sigma);
-
-      LDLT_factor<T_covar, Eigen::Dynamic, Eigen::Dynamic> ldlt_Sigma(Sigma);
-      check_ldlt_factor(function, "LDLT_Factor of Sigma", ldlt_Sigma);
-
-      check_size_match(function,
-                       "Size of random variable (rows y)", y.rows(),
-                       "Size of kernel scales (w)", w.size());
-      check_size_match(function,
-                       "Size of random variable", y.cols(),
-                       "rows of covariance parameter", Sigma.rows());
-      check_positive_finite(function, "Kernel scales", w);
-      check_finite(function, "Random variable", y);
-
-      if (y.rows() == 0)
-        return lp;
-
-      if (include_summand<propto>::value) {
-        lp += NEG_LOG_SQRT_TWO_PI * y.rows() * y.cols();
-      }
-
-      if (include_summand<propto, T_covar>::value) {
-        lp -= 0.5 * log_determinant_ldlt(ldlt_Sigma) * y.rows();
-      }
-
-      if (include_summand<propto, T_w>::value) {
-        lp += (0.5 * y.cols()) * sum(log(w));
-      }
-
-      if (include_summand<propto, T_y, T_w, T_covar>::value) {
-        Eigen::Matrix<T_w, Eigen::Dynamic, Eigen::Dynamic>
-          w_mat(w.asDiagonal());
-        Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic> yT(y.transpose());
-        lp -= 0.5 * trace_gen_inv_quad_form_ldlt(w_mat, ldlt_Sigma, yT);
-      }
-
-      return lp;
+      return multi_gp_lpdf<propto, T_y, T_covar, T_w>(y, Sigma, w);
     }
 
+    /**
+     * @deprecated use <code>multi_gp_lpdf</code>
+     */
     template <typename T_y, typename T_covar, typename T_w>
     inline
     typename boost::math::tools::promote_args<T_y, T_covar, T_w>::type
     multi_gp_log(const Eigen::Matrix<T_y, Eigen::Dynamic, Eigen::Dynamic>& y,
                  const Eigen::Matrix<T_covar, Eigen::Dynamic, Eigen::Dynamic>&
                  Sigma,
                  const Eigen::Matrix<T_w, Eigen::Dynamic, 1>& w) {
-      return multi_gp_log<false>(y, Sigma, w);
+      return multi_gp_lpdf<T_y, T_covar, T_w>(y, Sigma, w);
```

Comment:
For future reference (not this pull request), those types would've all been inferred without listing them explicitly.